### ISSUE TITLE
After 4.2 ResolveUsing can't handle nulls.

### ISSUE THREAD

**Issue by billrob** [AutoMapper/AutoMapper#1123]
Sorry, I can't get this to format everything as a code block, or even bolded.  If you run this below in a console app you will see the error.  There was another work item that seeming had to do with nullable of T.  This is unrelated to that issue.
```
**Mapping types:
Object -> DestinationInner
System.Object -> ConsoleApplication2.Program+DestinationInner

Destination path:
Destination.Inner.Inner

Source value:
(null)
   at AutoMapper.MappingEngine.Map(ResolutionContext context)
   at AutoMapper.Mappers.TypeMapObjectMapperRegistry.PropertyMapMappingStrategy
MapPropertyValue(ResolutionContext context, Object mappedObject, PropertyMap pr
pertyMap)**

`static void Main(string[] args)
{
	var mapper = new AutoMapper.MapperConfiguration(cfg =>
	{
		cfg.CreateMap<Source, Destination>()
			.ForMember(dest => dest.Inner, opt => opt.ResolveUsing(src => src.Inner));
		cfg.CreateMap<SourceInner, DestinationInner>();
	}).CreateMapper();

	var goodSource = new Source()
	{
		Inner = new SourceInner() { Name = "BillRob", },
	};
	var goodDestination = mapper.Map<Source, Destination>(goodSource);
	Console.WriteLine(goodDestination.Inner.Name);

	var nullSource = new Source();
	var nullDestination = mapper.Map<Source, Destination>(nullSource);
	//exception thrown.
}

public class Source
{
	public Source()
	{				
	}

	public SourceInner Inner { get; set; }
}

public class SourceInner
{
	public string Name { get; set; }
}

public class Destination
{
	public DestinationInner Inner { get; set; }
}

public class DestinationInner
{
	public string Name { get; set; }
}
```

**Comment by lbargaoanu** [comment]
Try it on the MyGet build.

**Comment by billrob** [comment]
I'm not sure how to pull the MyGet builds.  I found this announcement here.  https://lostechies.com/jimmybogard/2015/07/07/automapper-dev-builds-moved-to-myget/

**Comment by TylerCarlson1** [comment]
https://docs.nuget.org/consume/package-manager-dialog#package-sources

Follow rules and use the MyGet url for source.  Then you can search and update using your new package source
https://www.myget.org/F/automapperdev/api/v2

**Comment by slawomirbrys** [comment]
I have the same issue. When I use customer resolver class that inherits from `AutoMapper.ValueResolver` then everything works fine.

**Comment by jakubcermak** [comment]
I can confirm the issue.

**Comment by billrob** [comment]
@jakubcermak Did you confirm on the myget.org release?

**Comment by Tobbe1974** [comment]
I can confirm we have the same issue in our solution.
ResolveUsing can't handle nulls even though AllowNullCollections and AllowNullDestinationValues are set to true.
Trying to map from one ICollection<> to another.

**Comment by chihoyeung** [comment]
same here, using 4.2.1, works when using custom ValueResolver

**Comment by mcartoixa** [comment]
Same thing. If you use a custom ValueResolver configured with `FromMember` (which takes a `Expression<Func<TSource, object>>` parameter, notice the `object` type parameter) the bug is reproduced.
AutoMapper 4.2.1

**Comment by lbargaoanu** [comment]
Guys, this was fixed :) Try the MyGet build.

**Comment by mcartoixa** [comment]
@lbargaoanu I can't get any update from https://www.myget.org/F/automapperdev/api/v2. So I can't wait for the official (or beta) release to be out! :wink:

**Comment by jbogard** [comment]
And I'm just waiting on .NET Core RC2 to drop.

SOON.gif

**Comment by robsonj** [comment]
Hi There,

We're in the process of trying to move from v3.1.1 to v5. To ease the transition I am trying to first move us to v4.2.1, however our unit tests are now failing due to this bug in v4.2.1. Will you be creating a v4.2.2 with this fix please?

To move directly from v3.1.1 to v5 would be a huge leap for us without being able to complete this intermediate step of using the v4 vintage to move to the non-static API piece by piece.

Cheers
Jonathan

**Comment by jbogard** [comment]
4.2.1 and 5.0.2 have a static API.

And no, no plans for a 4.2.2 release.

**Comment by robsonj** [comment]
Aaa, so the static API got re-added on 5.0.2? and this fix is also in 5.0.2

**Comment by jbogard** [comment]
Technically it got added back (or un-marked obsolete) in 4.2.1
